### ISSUE TITLE
Tooltip on the select box still appearing after clear.

### ISSUE THREAD

**Issue by Aashishkmr** [select2/select2#3895]
I have a select box and i have used select2.js to create a select control out of it. My initialization code is :  $("#mycontrol).select2({  });

and i clear it like this $("#mycontrol").select2("val","");

Its getting cleared fine but the tooltip on the select2 dropdown control still displays the older value.
The same thing happens if i use the croos icon to clear the selected value. The tooltip still has the value same as last selected item.

**Comment by kevin-brown** [comment]
I can confirm this issue in the placeholder demo on the examples page when using 4.0.0.

**Comment by Aashishkmr** [comment]
Hi Kevin can you suggest a workaround for this. I have found that the tilte is bound on this span field
<span class="select2-selection__rendered" id="select2-ddMatterRequest-container"
</span>

One way can be finding this HTML control and removing title attribute from it. If you have something better please advise.

**Comment by fengdh** [comment]
A bugfix proposal:

```
  SingleSelection.prototype.clear = function () {
    this.$selection.find('.select2-selection__rendered').empty()
        .prop('title', '');
  };
```

**Comment by lenamtl** [comment]
+1

**Comment by scarabdesign** [comment]
It appears that the fix on Oct 17, 2016 addresses the issue if the "clear" method is called, not if the value of the selector is changed to null (and the change it triggered), so this still appears to be a standing issue.

**Comment by Gavin-Paolucci-Kleinow** [comment]
Has this issue been resolved yet? Are there any plans to do so?

**Comment by Gavin-Paolucci-Kleinow** [comment]
Bump to keep this issue open longer.

**Comment by kevin-brown** [comment]
As has been noted, the most obvious issue was fixed in 4.0.6-rc.0 where clicking the clear button didn't reset the tooltip.

> not if the value of the selector is changed to null (and the change it triggered)

I'm guessing this has something to do with us trying to set the title to `null` and jQuery ignoring that because the title can't be set to that?

---

I'm always willing to review pull requests that fix issues, especially ones with unit tests (otherwise I have to spend time writing said unit tests).
